lib/rucio/core/replica.py

```diff
@@ -664,40 +665,36 @@ def _list_replicas_for_datasets(dataset_clause, state_clause, rse_clause, sessio
     if rse_clause is not None:
         replica_query = replica_query.filter(or_(*rse_clause))
 
+    if updated_after:
+        replica_query = replica_query.filter(models.RSEFileAssociation.updated_at >= updated_after)
+
     for replica in replica_query.yield_per(500):
         yield replica
 
 
-def _list_replicas_for_files(file_clause, state_clause, files, rse_clause, session):
+def _list_replicas_for_files(file_clause, state_clause, files, rse_clause, updated_after, session):
     """
     List file replicas for a list of files.
 
     :param session: The database session in use.
     """
     for replica_condition in chunks(file_clause, 50):
 
-        if state_clause is None:
-            if rse_clause is None:
-                whereclause = and_(models.RSEFileAssociation.rse_id == models.RSE.id,
-                                   models.RSE.deleted == false(),
-                                   or_(*replica_condition))
-            else:
-                whereclause = and_(models.RSEFileAssociation.rse_id == models.RSE.id,
-                                   models.RSE.deleted == false(),
-                                   or_(*replica_condition),
-                                   or_(*rse_clause))
-        else:
-            if rse_clause is None:
-                whereclause = and_(models.RSEFileAssociation.rse_id == models.RSE.id,
-                                   models.RSE.deleted == false(),
-                                   state_clause,
-                                   or_(*replica_condition))
-            else:
-                whereclause = and_(models.RSEFileAssociation.rse_id == models.RSE.id,
-                                   models.RSE.deleted == false(),
-                                   state_clause,
-                                   or_(*replica_condition),
-                                   or_(*rse_clause))
+        filters = [models.RSEFileAssociation.rse_id == models.RSE.id,
+                   models.RSE.deleted == false(),
+                   or_(*replica_condition),
+                   ]
+
+        if state_clause is not None:
```

Comment:
any reason for the `is not None`? would it suffice to say `if state_clause:` ?

Reply:
no because clauses have no method to cast into boolean